src/test/java/org/javarosa/core/model/data/test/TimeDataLimitationsTest.java

```diff
@@ -0,0 +1,67 @@
+package org.javarosa.core.model.data.test;
+
+import org.javarosa.core.model.data.TimeData;
+import org.javarosa.core.model.utils.DateUtils;
+import org.junit.Test;
+
+import java.util.TimeZone;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ This test is intended to show the limitation of {@link TimeData}.
+
+ Using this data type in countries which change their time (summer/winter - DST) will cause that forms saved during
+ wintertime and then edited during summertime (and vice versa) will be treated as saved in a neighbor's timezone.
+ It's because we have just time and time offset like: 10:00:00.000+02:00 but we don't know when the form has been
+ saved so we parse it using the current date.
+
+ Example:
+ If we saved 10:00:00.000+02:00 during summertime (in Poland) and we are editing the form during winter time our
+ timezone is +01:00 not +02:00. As mentioned above javarosa doesn't know that the form has been saved in the same location
+ but different timezone because of DST so it treats the value like saved in the neighbor timezone
+ (in Kiev or London for example).
+
+ Related issues:
+ https://github.com/opendatakit/javarosa/pull/478
+ https://github.com/opendatakit/collect/issues/170
+ */
+public class TimeDataLimitationsTest {
+
```

Comment:
We should add a @Before and @After that ensures each test runs in isolation and has no side-effects for other tests.

```java
private TimeZone backupTimeZone;

@Before
public void setUp() {
    backupTimeZone = TimeZone.getDefault();
}

@After
public void tearDown() {
    TimeZone.setDefault(backupTimeZone);
}
```

Reply:
Fixed.